src/LaunchDarkly.ServerSdk.Redis/RedisBigSegmentStoreImpl.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using LaunchDarkly.Sdk.Server.Interfaces;
+using LaunchDarkly.Logging;
+using StackExchange.Redis;
+
+using static LaunchDarkly.Sdk.Server.Interfaces.BigSegmentStoreTypes;
+
+namespace LaunchDarkly.Sdk.Server.Integrations
+{
+    internal sealed class RedisBigSegmentStoreImpl : RedisStoreImplBase, IBigSegmentStore
+    {
+        private readonly string _syncTimeKey;
+        private readonly string _includedKeyPrefix;
+        private readonly string _excludedKeyPrefix;
+
+        internal RedisBigSegmentStoreImpl(
+            ConfigurationOptions redisConfig,
+            string prefix,
+            Logger log
+            ) : base(redisConfig, prefix, log)
+        {
+            _syncTimeKey = prefix + ":big_segments_synchronized_on";
+            _includedKeyPrefix = prefix + ":big_segment_include:";
+            _excludedKeyPrefix = prefix + ":big_segment_exclude:";
+        }
+
+        public async Task<IMembership> GetMembershipAsync(string userHash)
+        {
+            var db = _redis.GetDatabase();
+
+            var includedRefs = await db.SetMembersAsync(_includedKeyPrefix + userHash);
+            var excludedRefs = await db.SetMembersAsync(_excludedKeyPrefix + userHash);
+
+            return NewMembershipFromSegmentRefs(RedisValuesToStrings(includedRefs),
+                RedisValuesToStrings(excludedRefs));
+            throw new NotImplementedException();
+        }
+
+        public async Task<StoreMetadata?> GetMetadataAsync()
+        {
+            var db = _redis.GetDatabase();
+
+            var value = await db.StringGetAsync(_syncTimeKey);
+            if (value.IsNull)
+            {
+                return null;
+            }
+            if (value == "")
+            {
+                return new StoreMetadata { LastUpToDate = null };
+            }
+            var millis = long.Parse(value);
```

Comment:
I'm wondering about the edge case of this parse failing. Would it make more sense to treat this the same as the `value == ""` case above? If so the above case wouldn't need to be explicit as `""` would also fail the parse. Otherwise it looks like the SDK will consider the store unavailable for status updates, but the evaluation result won't be marked stale.

Reply:
The spec doesn't address this. I think I was making an assumption, based on the fact that `LastUpToDate` is an optional value in `StoreMetadata`, that there would have to be some non-erroneous way to represent such a state... but I think actually having `LastUpToDate` be an optional value was something I made up, or did by accident, and not something that was in the spec. I don't know if we want to have a way to indicate "ld-relay has touched this database to indicate that it's alive, but hasn't done a big segment sync yet"; @hroederld, can you think of any reason that would be relevant?

Reply:
After some Slack conversation, this implementation makes sense. An invalid value for the last synchronized date should be treated as a store error. One thing I was missing before is that returning `new StoreMetadata { LastUpToDate = null }` is treated the same as returning `null` in this context. Despite the fact that the current implementation of the relay proxy will not set an empty string value for this key, it seems reasonable to treat that as the data being potentially stale rather than invalid.